### PR TITLE
[BEACON-444] [BEACON-446] Cleanup and introduce ES6 with classes

### PR DESCRIPTION
**Pull request by williamrijksen**
### Ticket
https://enrise.atlassian.net/browse/BEACON-444.
https://enrise.atlassian.net/browse/BEACON-446.

### What has been done
 - Cleanup code
 - Introduce classes for the scanners.
 - Check the module pathsense / iBeacons is included and decides which strategy is used.

### How to test
 - Add the latests dist to the fietsnetwerk app and verify nothing has been broken.

### Todo
  - [x] Introduce Travis for ESLint
  - [x] Make the BLE and location check optional.

### Notes
Blocked by #20.

### CODE REVIEW

File: lib-ng/sensimity.js [hunk 1]
```diff
@@ -37,12 +37,18 @@ const startScanner = options => {
...
-const start = (options, callback) =>
+const start = (args, callback) =>
   // Only start Sensimity when bluetooth is enabled
   requestLocationPermissions(
     (e) => e.success
     ? isBLEEnabled(isEnabled => {
-      if (!isEnabled) {
+      const options = Object.assign({
+        requireBLE: true,
+        startGeofence: true,
```
Comment (jvandijk) [inline lib-ng/sensimity.js]: should we name it `scanGeofence`?

File: lib-ng/sensimity.js [hunk 2]
```diff
@@ -1,214 +1,166 @@
...
-    Alloy.Globals.sensimityEvent = require('./dispatcher');
-}
+// Create an scanner, specific for the running platform
+const startScanner = options => {
+  if (!_.isUndefined(Alloy.Globals.sensimityScan) || !_.has(options, 'networkId')) {
+    return;
+  }
+
+  let scanOptions = options;
+  if (Ti.Platform.name === 'android') {
```
Comment (williamrijksen) [inline lib-ng/sensimity.js]: Set to `osname` please

File: lib-ng/scanners/AltbeaconScanner.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          foregroundBetweenScanPeriod: 0,
+          backgroundScanPeriod: 2001,
+          backgroundBetweenScanPeriod: 5001,
+        },
+      };
+      this.runInService = runInService;
+      this.beaconFound = this.beaconFound.bind(this);
+      this.addAllEventListeners();
+    } catch (e) {
+      Ti.API.warn('Could not start BLE-scan, please include the com.drtech.altbeacon module');
```
Comment (williamrijksen) [inline lib-ng/scanners/AltbeaconScanner.js]: please change to `ti.api.info`

**Comment by jvandijk** [comment]
Introduce what?

**Comment by jvandijk** [comment]
Merging for follow up PR

**Pull request merged**